libraries/AP_AHRS/AP_AHRS_NavEKF.cpp

```diff
@@ -1671,30 +1671,32 @@ bool AP_AHRS_NavEKF::attitudes_consistent(char *failure_msg, const uint8_t failu
 #endif
 
     // check primary vs dcm
-    Quaternion dcm_quat;
-    Vector3f angle_diff;
-    dcm_quat.from_rotation_matrix(get_DCM_rotation_body_to_ned());
-    primary_quat.angular_difference(dcm_quat).to_axis_angle(angle_diff);
-    float diff = safe_sqrt(sq(angle_diff.x)+sq(angle_diff.y));
-    if (diff > ATTITUDE_CHECK_THRESH_ROLL_PITCH_RAD) {
-        hal.util->snprintf(failure_msg, failure_msg_len, "DCM Roll/Pitch inconsistent by %d deg", (int)degrees(diff));
-        return false;
-    }
-
-    // we only check yaw against DCM if we are not using external yaw
-    // for EKF3. DCM can't use external yaw, so we don't expect it's
-    // yaw to align with EKF3 when EKF3 is using an external yaw
-    // source
-    bool using_external_yaw = false;
-#if HAL_NAVEKF3_AVAILABLE
-    using_external_yaw = ekf_type() == EKFType::THREE && EKF3.using_external_yaw();
-#endif
-    if (!using_external_yaw) {
-        diff = fabsf(angle_diff.z);
-        if (check_yaw && (diff > ATTITUDE_CHECK_THRESH_YAW_RAD)) {
-            hal.util->snprintf(failure_msg, failure_msg_len, "DCM Yaw inconsistent by %d deg", (int)degrees(diff));
+    if (!always_use_EKF()) {
```

Comment:
only skip DCM if only single EKF lane?

Reply:
i've made the change so it only checks DCM if the vehicle might use DCM (i.e. always_use_EKF is false) OR if there is only a single EKF core being used.